pkg/resolve/resolve.go

```diff
@@ -15,18 +15,48 @@ import (
 	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
 	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	k8scheme "k8s.io/client-go/kubernetes/scheme"
+	yaml "sigs.k8s.io/yaml/goyaml.v2"
 )
 
 type TektonTypes struct {
-	PipelineRuns []*tektonv1.PipelineRun
-	Pipelines    []*tektonv1.Pipeline
-	TaskRuns     []*tektonv1.TaskRun
-	Tasks        []*tektonv1.Task
+	PipelineRuns     []*tektonv1.PipelineRun
+	Pipelines        []*tektonv1.Pipeline
+	TaskRuns         []*tektonv1.TaskRun
+	Tasks            []*tektonv1.Task
+	ValidationErrors map[string]string
+}
+
+func NewTektonTypes() TektonTypes {
+	return TektonTypes{
+		ValidationErrors: map[string]string{},
+	}
 }
 
 var yamlDocSeparatorRe = regexp.MustCompile(`(?m)^---\s*$`)
 
+// detectAtleastNameOrGenerateNameFromPipelineRun detects the name or
+// generateName of a yaml files even if there is an error decoding it as tekton types.
+func detectAtleastNameOrGenerateNameFromPipelineRun(data string) string {
+	var metadataName struct {
+		Metadata metav1.ObjectMeta
+	}
+	err := yaml.Unmarshal([]byte(data), &metadataName)
+	if err != nil {
+		return ""
+	}
+	if metadataName.Metadata.Name != "" {
+		return metadataName.Metadata.Name
+	}
+
+	// TODO: yaml Unmarshal don't want to parse generatename and i have no idea why
```

Comment:
how about writing own Metadata in the above struct ??
```
	var metadataName struct {
		Metadata struct {
			GenerateName string `yaml:"generateName"`
			Name         string `yaml:"name"`
		} `yaml:"metadata"`
	}
```
this way we can handle generateName also :thinking: